### PR TITLE
Unit tests FTW

### PR DESCRIPTION
**Pull request by JPLachance**
Add unit tests and make the coverage report work on Typescript files!

Also fix a couple of issues I found and update outdated libs!
screen shot 2016-09-22 at 5 48 49 pm

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,38 +23,59 @@
...
     "prepublish": "npm run build"
   },
   "dependencies": {
     "react-tether": "0.5.2"
   },
   "peerDependencies": {
     "tether": "^1.3.7",
-    "react": "^15.3.1",
-    "react-dom": "^15.3.1"
+    "react": "^15.3.2",
```
Comment (pochretien) [inline package.json]: remove ^

File: tests/components/PopoverComponent.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          >
+            <span id='PopoverToggle'>Toggle</span>
+            {false}
+          </PopoverComponent>
+        </div>,
+        {attachTo: document.getElementById('App')}
+      );
+    });
+
+    it('should not explode', () => {
```
Comment (pochretien) [inline tests/components/PopoverComponent.spec.tsx]: why this test? its a duplicate i think

**Comment by wfortin** [comment]
All good !

File: package.json (hunk 1 above)
Comment (JPLachance) [inline package.json]: https://nodejs.org/en/blog/npm/peer-dependencies/

So we don't want to force users of our library to use a fixed version of react, we only want to warn them on the fact they need to install it and they need to have at least the version specified there.

File: tests/components/PopoverComponent.spec.tsx (hunk 2 above)
Comment (JPLachance) [inline tests/components/PopoverComponent.spec.tsx]: Like said in the describe: Document click handler without PopoverElement in DOM

In the test before, the popover had a tether element, but here it doesn't. That's a small difference in the code that we need to cover.

File: src/components/PopoverComponent.tsx [hunk 3]
```diff
@@ -8,6 +8,15 @@ export interface IPopoverComponentProps extends TetherComponent.ITetherComponent
 }
 
 export class PopoverComponent extends React.Component<IPopoverComponentProps, any> {
+  static propTypes = _.extend(TetherComponent.propTypes, {
+    children: ({children}, propName, componentName) => {
+      if (_.isUndefined(children) || React.Children.count(children) != 2) {
+        return new Error(`${componentName} expects two children to use as target and element.` +
+          `Second child can either be a boolean or a ReactNode.`);
```
Comment (GermainBergeron) [inline src/components/PopoverComponent.tsx]: No need to end your string and put a# `+` with the string literals.

File: tests/components/PopoverComponent.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+      toggleOpenedTetherElement: toggleOpenedTetherElementSpy
+    };
+  });
+
+  it('should render without error', () => {
+    expect(() => {
+      shallow(
+        <PopoverComponent
+          {...popoverComponentProps}
+        >
```
Comment (GermainBergeron) [inline tests/components/PopoverComponent.spec.tsx]: IMO it looks weird with the `>` alone on its line

File: tsconfig.test.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    "typings/globals/marionette/index.d.ts",
+    "typings/globals/react-dom/index.d.ts",
+    "typings/globals/react-redux/index.d.ts",
+    "typings/globals/react-tether/index.d.ts",
+    "typings/globals/react/index.d.ts",
+    "typings/globals/redux/index.d.ts",
+    "typings/globals/tether/index.d.ts",
+    "typings/globals/underscore/index.d.ts",
+    "typings/globals/webpack-env/index.d.ts",
+    "typings/index.d.ts",
```
Comment (GermainBergeron) [inline tsconfig.test.json]: I think you should just have `"typings/index.d.ts"` instead of all the files in the `typings` folder

Comment (wfortin) [inline tsconfig.test.json]: +1

File: src/components/PopoverComponent.tsx (hunk 3 above)
Comment (JPLachance) [inline src/components/PopoverComponent.tsx]: OK, but the formatting kinda suck...
```
  return new Error(`${componentName} expects two children to use as target and element. Second child can either be a boolean or a ReactNode.`);
```

File: tsconfig.test.json (hunk 5 above)
Comment (JPLachance) [inline tsconfig.test.json]: I did the migration to TS 2.0 in another branch, no need for that anymore :)

File: tests/components/PopoverComponent.spec.tsx (hunk 4 above)
Comment (JPLachance) [inline tests/components/PopoverComponent.spec.tsx]: Fixed!

**Pull request merged**